core/extensions/impl/storage_extension.hpp

```diff
@@ -173,10 +173,15 @@ namespace kagome::extensions {
 
     outcome::result<common::Buffer> get(const common::Buffer &key) const;
 
-    outcome::result<boost::optional<common::Buffer>> getStorageNextKey(
-        const common::Buffer &key) const;
+    /**
+     * @return error if any, a pair with (opt_key, true)
+     * if \arg key is found, (none, false) otherwise
+     */
+    outcome::result<std::pair<boost::optional<common::Buffer>, bool>>
```

Comment:
Do we need to return bool? Just optional should be enough. From what I see in implementation bool value is always true when buffer value exists